From 55b031b75165cf5d8302ccf1da71ae0a5fc7edaf Mon Sep 17 00:00:00 2001
From: Benjamin Reis <benjamin.reis@vates.tech>
Date: Wed, 18 Sep 2024 15:21:29 +0200
Subject: [PATCH] Fix network reset script in static IPv6

Only exits script when an error occurs when
getting the static IPv6 config info

Signed-off-by: Benjamin Reis <benjamin.reis@vates.tech>
---
 scripts/xe-reset-networking | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/scripts/xe-reset-networking b/scripts/xe-reset-networking
index a5bd437f9d..3ce5d55a79 100755
--- a/scripts/xe-reset-networking
+++ b/scripts/xe-reset-networking
@@ -141,10 +141,11 @@ if __name__ == "__main__":
   
 	if options.mode_v6 == 'static':
 		if options.ipv6 == '':
-	 		parser.error("if static IPv6 mode is selected, an IPv6 address needs to be specified")
-		elif options.ipv6.find('/') == -1:
+			parser.error("if static IPv6 mode is selected, an IPv6 address needs to be specified")
+			sys.exit(1)
+		if options.ipv6.find('/') == -1:
 			parser.error("Invalid format: IPv6 must be specified with CIDR format: <IPv6>/<prefix>")
-		sys.exit(1)
+			sys.exit(1)
   
 	# Warn user
 	if not os.access('/tmp/fist_network_reset_no_warning', os.F_OK):
